File: src/main/java/model/OrderRow.java
package model;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.persistence.Table;
import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.NotNull;


@Data
@AllArgsConstructor
@NoArgsConstructor
@Embeddable
@Table(name = "order_rows")
public class OrderRow {

    @NotNull
    @Column(name = "item_name")
    public String itemName;

    @NotNull
    @DecimalMin("1")
    public Integer quantity;

    @NotNull
    @DecimalMin("1")
    public Integer price;
}

File: src/main/kotlin/model/OrderRow.kt
package model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.NotNull


@JsonIgnoreProperties("id", "orderId")
data class OrderRow (
    var id: Long? = null,
    @get:NotNull var itemName:  String? = null,
    @get:NotNull @get:DecimalMin("1") var quantity:  Int? = null,
    @get:NotNull @get:DecimalMin("1") var price:  Int? = null,
    var orderId: Long? = null
)